lib/cli.ex

```diff
@@ -1,7 +1,20 @@
 defmodule Onigumo.CLI do
-  def main([component]) do
-    module = Module.safe_concat("Onigumo", component)
+  def main(argv) do
+    {_parsed, args, _invalid} = OptionParser.parse(argv, strict: [])
+
+    module =
+      args
+      |> get_component
+      |> safe_concat
```

Comment:
After a productive discussion, we abandoned _safe_concat_ in favor of a [module literal](https://github.com/Glutexo/onigumo/blob/f7272069fcc32abdc4ddc6333fae150d657c0f08/lib/cli.ex#L3).